Remove interactive runner

camillobruni opened the pull request:
The interactive runner has low added values these days:
- We can easily run individual workloads via the developerMode menu
- The control-flow in the async-await-based runner is much easier to follow and thus it's easier to set the right breakpoints

To simplify future runner refactorings I suggest removing the interactive runner support all together (and if we need some additional support, we should expose that in the developerMode)

bgrins reviewed (approved):
Works for me, as discussed. I do find this somewhat convenient at times when developing tests, but we can find ways to simulate it with individual steps or perhaps in developer mode if we find it necessary to re-add.

julienw reviewed (approved):
I agree with removing it, but it would be good to have a replacement for the "step by step" way of running tests. I found this mode valuable when I developed my tests last year.

camillobruni commented:
Can't you easily use breakpoints for step-by-step debugging in the current runner (in `_runTestAndRecordResults`)?

rniwa commented:
It would be useful to have a step-by-step mode. I use it all the time while debugging Speedometer content.

camillobruni commented:
Ok, let me add a "Step Workload" button in the devMode menu, that shouldn't be too hard to implement.

camillobruni commented:
Updated the code and included a stepping button and functionality, please have a look.
I've now added the promise wait on the `willRunTest` client hook as it seemed more natural, but maybe the `didRunTest` approach as in the current interactive runner is better?

rniwa commented on resources/main.mjs:
```diff
@@ -17,17 +17,49 @@ class MainBenchmarkClient {
+    _clearStepping() {
+        this._steppingResolver();
+        this._steppingPromise = null;
+        this._steppingResolver = null;
```
Shouldn't we be clearing these two states before calling `_steppingResolver`?

camillobruni replied:
sounds good.

rniwa commented on resources/developer-mode.mjs:
```diff
@@ -222,14 +222,20 @@ function createUIForSuites() {
-    button.textContent = "Start Test";
-    button.onclick = (event) => {
+    const stepTestButton = document.createElement("button");
+    stepTestButton.textContent = "Step Test ⏯";
```
Can we use `&#9199;` or '\u23EF' instead of directly using the unicode character?

camillobruni replied:
done.

rniwa commented on resources/developer-mode.mjs:
```diff
@@ -222,14 +222,20 @@ function createUIForSuites() {
+        globalThis.benchmarkClient.step();
+    };
+    const startTestButton = document.createElement("button");
+    startTestButton.textContent = "Start Test ⏵";
```
Ditto for `&#9205;` or `\u23F5`
